src/cirq_iqm/extended_qasm_parser.py

```diff
@@ -15,20 +15,39 @@
 Extends the OpenQASM 2.0 language by gates native to the IQM architectures.
 """
 import cirq
-from cirq.contrib.qasm_import._parser import QasmParser
+import numpy as np
+from cirq.contrib.qasm_import._parser import (QasmGateStatement, QasmParser,
+                                              QasmUGate)
 
 
 def circuit_from_qasm(qasm: str) -> cirq.circuits.Circuit:
     """Parses an OpenQASM 2.0 program to a Cirq circuit.
 
-   TODO The previous functionality that contained additional 2-qubit gates has now been removed as obsolete.
-        This function remains because new parsing logic will be introduced in a future PR.
-
     Args:
-        qasm: OpenQASM string
+        qasm: OpenQASM program
 
     Returns:
         parsed circuit
     """
     parser = QasmParser()
+
+    def convert_U(args: tuple) -> cirq.Gate:
+        """Maps the OpenQASM builtin one-qubit gate ``U`` to :class:`cirq.QasmUGate`,
+        or :class:`cirq.PhasedXPowGate` if possible.
+        """
+        theta, phi, lambd = args
```

Comment:
```suggestion
        theta, phi, lambda = args
```

Reply:
Minor

Reply:
`lambda` is a reserved word in Python and so it can't be used as a variable name. I think `lambd` is fine.

Reply:
Ah yes 🙂 In that case I suggest to use the same abbreviation as `QasmUGate` uses: `lmda`